packages/swapper/src/swappers/ChainflipSwapper/swapperApi/getTradeQuote.ts

```diff
@@ -0,0 +1,330 @@
+import { Err, Ok } from '@sniptt/monads'
+import type { AxiosError } from 'axios'
+import { v4 as uuid } from 'uuid'
+
+import { getDefaultSlippageDecimalPercentageForSwapper } from '../../../constants'
+import type {
+  GetTradeQuoteInput,
+  GetUtxoTradeQuoteInput,
+  SwapErrorRight,
+  SwapperDeps,
+  TradeQuote,
+} from '../../../types'
+import {
+  type GetEvmTradeQuoteInput,
+  type ProtocolFee,
+  SwapperName,
+  TradeQuoteError,
+} from '../../../types'
+import { getRate, makeSwapErrorRight } from '../../../utils'
+import {
+  CHAINFLIP_BAAS_COMMISSION,
+  CHAINFLIP_BOOST_SWAP_SOURCE,
+  CHAINFLIP_DCA_BOOST_SWAP_SOURCE,
+  CHAINFLIP_DCA_QUOTE,
+  CHAINFLIP_DCA_SWAP_SOURCE,
+  CHAINFLIP_REGULAR_QUOTE,
+  CHAINFLIP_SWAP_SOURCE,
+  chainIdToChainflipNetwork,
+  usdcAsset,
+} from '../constants'
+import type { ChainflipBaasQuoteQuote, ChainflipBaasQuoteQuoteFee } from '../models'
+import { chainflipService } from '../utils/chainflipService'
+import { getEvmTxFees } from '../utils/getEvmTxFees'
+import { getUtxoTxFees } from '../utils/getUtxoTxFees'
+import { isSupportedAssetId, isSupportedChainId } from '../utils/helpers'
+
+export const getTradeQuote = async (
+  input: GetTradeQuoteInput,
+  deps: SwapperDeps,
+): Promise<Result<TradeQuote[], SwapErrorRight>> => {
+  const {
+    sellAsset,
+    buyAsset,
+    accountNumber,
+    receiveAddress,
+    sellAmountIncludingProtocolFeesCryptoBaseUnit: sellAmount,
+    affiliateBps: commissionBps,
+  } = input
+
+  if (accountNumber === undefined) {
+    return Err(
+      makeSwapErrorRight({
+        message: `accountNumber is required`,
+        code: TradeQuoteError.UnknownError,
+      }),
+    )
+  }
+
+  if (!isSupportedChainId(sellAsset.chainId)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `unsupported chainId`,
+        code: TradeQuoteError.UnsupportedChain,
+        details: { chainId: sellAsset.chainId },
+      }),
+    )
+  }
+
+  if (!isSupportedChainId(buyAsset.chainId)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `unsupported chainId`,
+        code: TradeQuoteError.UnsupportedChain,
+        details: { chainId: sellAsset.chainId },
+      }),
+    )
+  }
+
+  if (!isSupportedAssetId(sellAsset.chainId, sellAsset.assetId)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `asset '${sellAsset.name}' on chainId '${sellAsset.chainId}' not supported`,
+        code: TradeQuoteError.UnsupportedTradePair,
+        details: { chainId: sellAsset.chainId, assetId: sellAsset.assetId },
+      }),
+    )
+  }
+
+  if (!isSupportedAssetId(buyAsset.chainId, buyAsset.assetId)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `asset '${buyAsset.name}' on chainId '${buyAsset.chainId}' not supported`,
+        code: TradeQuoteError.UnsupportedTradePair,
+        details: { chainId: buyAsset.chainId, assetId: buyAsset.assetId },
+      }),
+    )
+  }
+
+  const sellChainflipChainKey = `${sellAsset.symbol.toLowerCase()}.${
```

Comment:
Another fyi, if you want to improve this. The assets are available here: https://chainflip-broker.io/assets as the `id` property in the format `asset`.`chain`, it just so happened all of them matched the `.symbol`, but perhaps you want to have this in a mapping in constants as well, like the chains?

Reply:
I added this to the axios cache by the way, if it helps :) Didn't use it though

Reply:
Think that's fine as a first pass IMO given there aren't many assets supported, but as Chainflip supports more assets, we'll definitely want to make things programmatic by hitting this endpoint so we don't have to manually add supported assets everytime!